lib/typed.js

```diff
@@ -337,6 +337,26 @@
         var ch, ch2;
 
         value = advance();
+
+        // check for import("./path/to/module").Foo type
+        var importStart = 'import(';
+        if (source.slice(index - 1, index - 1 + importStart.length) === importStart) {
+            value = importStart;
+            index += importStart.length - 1;
+            // scan until closing )
+            while (index < length && source.charAt(index) !== ')') {
+                value += advance();
+            }
+            if (index >= length) {
+                return Token.ILLEGAL;
+            }
+            value += advance();
+
+            if (source.charCodeAt(index) !== 0x2E  /* '.' */) {
+                return Token.NAME;
+            }
+        }
+
         while (index < length && isTypeName(source.charCodeAt(index))) {
             ch = source.charCodeAt(index);
             if (ch === 0x2E  /* '.' */) {
```

Comment:
This loop handles what Typescript calls EntityNames: `x.y.z.a`, for example. It breaks when it sees a `<`, which makes me think that some other code handles type arguments. It would be nice to have a test with type arguments, like `import("foo").SortedSet<number>`.